packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -133,6 +133,7 @@ class TimeInput extends React.Component {
     this.handleMeridiemInputFocus = this.handleMeridiemInputFocus.bind(this);
     this.handleMeridiemSelectFocus = this.handleMeridiemSelectFocus.bind(this);
     this.handleMeridiemButtonChange = this.handleMeridiemButtonChange.bind(this);
+    this.doesComponentNeedsUpdate = this.doesComponentNeedsUpdate.bind(this);
```

Comment:
Let's not call it that. Let's just move the logic inside `componentDidUpdate` and there's no harm, imo.

Reply:
componentdidUpdate() looks complex if I move this in like below

```javascript
componentDidUpdate(prevProps, prevState) {
  let doesComponentNeedsUpdate = false;
  let { meridiem } = this.state;
  let hour = TimeUtil.splitHour(this.props.value);
  if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
    if (!this.props.intl.messages['Terra.timeInput.am'] || !this.props.intl.messages['Terra.timeInput.pm']) {
        if (process.env !== 'production') {
          // eslint-disable-next-line no-console
          console.warn('This locale only uses 24 hour clock. The ante meridiem and post meridiem will not be displayed');
        }

        this.anteMeridiem = '';
        this.postMeridiem = '';
      } else {
        this.anteMeridiem = this.props.intl.formatMessage({ id: 'Terra.timeInput.am' });
        this.postMeridiem = this.props.intl.formatMessage({ id: 'Terra.timeInput.pm' });
      }

      if (hour) {
      const parsedHour = TimeUtil.parseTwelveHourTime(hour, this.anteMeridiem, this.postMeridiem);
      hour = parsedHour.hourString;
      // eslint-disable-next-line prefer-destructuring
      meridiem = parsedHour.meridiem;
    }
  }
  const minute = TimeUtil.splitMinute(this.props.value);
  const second = TimeUtil.splitSecond(this.props.value);

  const prevStateTimeValue = (this.props.showSeconds) ? `${prevState.hour}:${prevState.minute}:${prevState.second}` : `${prevState.hour}:${prevState.minute}`;
    // Comparing partial time values on each update will result in incorrect comparisons of prevState and propsValue.
    // Hence validating propsValue and prevState values to skip the comparison between them until time value is completely formed.
    if (TimeUtil.validateTime(this.props.value, this.props.showSeconds) && TimeUtil.validateTime(prevStateTimeValue, this.props.showSeconds)) {
      if (hour !== prevState.hour || minute !== prevState.minute || second !== prevState.second) {
        doesComponentNeedsUpdate = true;
    }
  }

  if (this.props.value === prevProps.value
    && this.props.variant === prevProps.variant && !doesComponentNeedsUpdate
  ) {
    return;
  }

   // eslint-disable-next-line react/no-did-update-set-state
   this.setState({
     hour,
     minute,
     second,
     meridiem,
   });
}
```

all these value assignments, conversions can be skipped if we have this condition before them. hence created new method to compare props and prevState.

Reply:
imo, this is not complex, and invoking a method named `doesComponentNeedsUpdate` inside this lifecycle method sounds a bit contradictory to me. As far as I understand from the updated code, it looks like you're checking if an update happened and not if an update needs to happen. This method name also confuses with the lifecycle method - `shouldComponentUpdate`.

Reply:
I agree with your suggestion. the main reason to add method `handleMissinghour()` for this condition is to relate the reason for which these changes are added. I.e. [this](https://github.com/cerner/terra-framework/blob/4ee766a28a07578ea5a76a52bf85cdfded0516f7/packages/terra-time-input/src/TimeInput.jsx#L235) check is will serve it's purpose only when TimeInput is used with DateTimePicker and adds no value for TimeInput when used as individual component. 

Updated :  996f20d7bf9a9ecbab8a126deebe87d22cd72259

Reply:
@nramamurth , @supreethmr  so with new componentupdate we are exiting after execution of code which is necessary only when we need to update the state. we could actually rename this old function **doesComponentNeedsUpdate** and use the previous logic of exiting this method at the beginning itself.

Reply:
yes that was a one of the reason for having method earlier. I believe moving this code to method improves readability.
would like to know your thoughts on this : @nramamurth

Reply:
I agree that this could return way sooner than it does now - I am not a big fan of adding a new method for it but maybe you could consider doing something similar. 

<details><summary>return sooner (also have a method (kinda))</summary>
<p>

```javascript
componentDidUpdate(prevProps, prevState) {
  let { meridiem } = this.state;
  let hour = TimeUtil.splitHour(this.props.value);
  const minute = TimeUtil.splitMinute(this.props.value);
  const second = TimeUtil.splitSecond(this.props.value);

  const shouldUpdateState = (() => {
    const prevStateTimeValue = (this.props.showSeconds) ? `${prevState.hour}:${prevState.minute}:${prevState.second}` : `${prevState.hour}:${prevState.minute}`;
    if (TimeUtil.validateTime(this.props.value, this.props.showSeconds) && TimeUtil.validateTime(prevStateTimeValue, this.props.showSeconds)) {
      return (hour !== prevState.hour || minute !== prevState.minute || second !== prevState.second);
    }
  })();

  if (!shouldUpdateState && this.props.value === prevProps.value && this.props.variant === prevProps.variant) {
    return;
  }

  if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
    if (!this.props.intl.messages['Terra.timeInput.am'] || !this.props.intl.messages['Terra.timeInput.pm']) {
        if (process.env !== 'production') {
          // eslint-disable-next-line no-console
          console.warn('This locale only uses 24 hour clock. The ante meridiem and post meridiem will not be displayed');
        }

        this.anteMeridiem = '';
        this.postMeridiem = '';
      } else {
        this.anteMeridiem = this.props.intl.formatMessage({ id: 'Terra.timeInput.am' });
        this.postMeridiem = this.props.intl.formatMessage({ id: 'Terra.timeInput.pm' });
      }

      if (hour) {
      const parsedHour = TimeUtil.parseTwelveHourTime(hour, this.anteMeridiem, this.postMeridiem);
      hour = parsedHour.hourString;
      // eslint-disable-next-line prefer-destructuring
      meridiem = parsedHour.meridiem;
    }
  }

   // eslint-disable-next-line react/no-did-update-set-state
   this.setState({
     hour,
     minute,
     second,
     meridiem,
   });
}
```
</p>
</details>

Reply:
as you mentioned above : 
```javascript
 const shouldUpdateState = (() => {
    const prevStateTimeValue = (this.props.showSeconds) ? `${prevState.hour}:${prevState.minute}:${prevState.second}` : `${prevState.hour}:${prevState.minute}`;
    if (TimeUtil.validateTime(this.props.value, this.props.showSeconds) && TimeUtil.validateTime(prevStateTimeValue, this.props.showSeconds)) {
      return (hour !== prevState.hour || minute !== prevState.minute || second !== prevState.second);
    }
  })();
```
we still need ```parsedHour``` for comparison of 12-hour format Time values and parsedHour requires :
 ```Javascript
  if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
    if (!this.props.intl.messages['Terra.timeInput.am'] || !this.props.intl.messages['Terra.timeInput.pm']) {
        if (process.env !== 'production') {
          // eslint-disable-next-line no-console
          console.warn('This locale only uses 24 hour clock. The ante meridiem and post meridiem will not be displayed');
        }

        this.anteMeridiem = '';
        this.postMeridiem = '';
      } else {
        this.anteMeridiem = this.props.intl.formatMessage({ id: 'Terra.timeInput.am' });
        this.postMeridiem = this.props.intl.formatMessage({ id: 'Terra.timeInput.pm' });
      }

      if (hour) {
      const parsedHour = TimeUtil.parseTwelveHourTime(hour, this.anteMeridiem, this.postMeridiem);
      hour = parsedHour.hourString;
      // eslint-disable-next-line prefer-destructuring
      meridiem = parsedHour.meridiem;
    }
  }
```
So to reduces the changes to be made in componentDidUpdate() created method : [hasTimeValueUpdated](https://github.com/cerner/terra-framework/blob/ef448d8d9d629e0543cdc606e5e855e39df22666/packages/terra-time-input/src/TimeInput.jsx#L231). 

As of now i am not able to see any impact of creating new method will remove method if I found any.